exonum/src/events/codec.rs

```diff
@@ -147,6 +163,39 @@ mod test {
         assert!(responder.decode(&mut bytes).is_err());
     }
 
+    #[test]
+    #[should_panic(expected = "Received malicious message with wrong total_len")]
+    fn decode_message_wrong_length() {
+        let data = vec![0u8, 0, 0, 0, 0, 0, 32, 0, 0, 0];
+        let mut bytes: BytesMut = BytesMut::new();
+        let (ref mut responder, ref mut initiator) = create_encrypted_codecs();
+        let raw = RawMessage::new(MessageBuffer::from_vec(data.clone()));
+        initiator.encode(raw, &mut bytes).unwrap();
+
+        responder.decode(&mut bytes).unwrap();
```

Comment:
I'd rearrange the code blocks a bit to make the test flow easier to follow:

```rust
let (ref mut responder, ref mut initiator) = create_encrypted_codecs();

let data = vec![0u8, 0, 0, 0, 0, 0, 32, 0, 0, 0];
let raw = RawMessage::new(MessageBuffer::from_vec(data));

let mut bytes: BytesMut = BytesMut::new();
initiator.encode(raw, &mut bytes).expect("encoding message");

responder.decode(&mut bytes).expect("decoding message");
```

There are people who are not fans of empty lines and promote terser code (this is only my personal preference), but I think that at least variable declarations could be regrouped and moved closer to their usage sites.